contracts/EthProxy.sol

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.6.7;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "./IUSM.sol";
+import "./external/IWETH9.sol";
+
+
+contract EthProxy {
+    using Address for address payable;
+    IUSM public usm;
+    IWETH9 public weth;
+
+    constructor(address usm_, address weth_) public {
+        usm = IUSM(usm_); 
+        weth = IWETH9(weth_);
+
+        weth.approve(address(usm), uint(-1));
+    }
+
+    /// @dev The WETH9 contract will send ether to EthProxy on `weth.withdraw` using this function.
+    receive() external payable { }
+
+    /// @dev Users use `mint` in EthProxy to post ETH to USM (amount = msg.value), which will be converted to Weth here.
+    /// @param to Receiver of the minted USM
+    function mint(address to)
+        external payable {
+        weth.deposit{ value: msg.value }();
+        usm.mint(address(this), to, msg.value);
+    }
```

Comment:
If it's useful for `USM.mint()` to return the amount minted, might it not also be useful for `EthProxy.mint()` to return the same (ie, `usm.mint()`)?  Seems like regardless of whether they both formally implement the same interface, we might as well give them the same API.

I also wonder if names like `WethUsm` (or `WethUSM` or something, dunno) and `EthUsm` might be clearer than `USM` and `EthProxy`.

Reply:
Fixed the return values.

Reply:
`EthProxy` will actually evolve towards just `Proxy`. In `USM.sol` we will have the core functions that allow USM to implement its business logic, in `Proxy.sol` all the functions that make it more convenient for users.

Reply:
Well I suppose it depends whether we want names that reflect how it works internally, or what service it provides to users.  From a user's pov, they're just two similar contracts, one takes ETH the other WETH.  Anyway my larger point was that EthProxy (and even more so Proxy) is a very generic non-USM-specific name: in other languages it would raise some risk of name conflicts with other libraries etc, but maybe in Solidity that's not a concern.

Reply:
The name code is not a concern. A conflict would happen only if we would have another contract with the same name. It doesn't matter if someone else chooses the same name for their contract.

On function names, I can append `FromEth` and `ToEth` to the methods in `Proxy`, if that helps. However, I expect that someone calling methods manually should notice that one contract is `USM` and the other `Proxy`, and be able to read the documentation to know what each function does.

Reply:
No the function names are fine.  They _should_ be the same!

Reply:
Should I rename `EthProxy` as `Proxy` already?

Reply:
I defer completely to you (guys) on this one!